crates/bevy_ui/src/flex/mod.rs

```diff
@@ -1,25 +1,32 @@
 mod convert;
 
-use crate::{CalculatedSize, Node, Style, UiScale};
+use crate::{prelude::UiRootCamera, CalculatedSize, Node, Style, UiScale};
 use bevy_ecs::{
     entity::Entity,
     event::EventReader,
-    query::{Changed, With, Without, WorldQuery},
+    query::{Changed, Or, With, Without, WorldQuery},
     system::{Query, RemovedComponents, Res, ResMut, Resource},
 };
 use bevy_hierarchy::{Children, Parent};
 use bevy_log::warn;
-use bevy_math::Vec2;
+use bevy_math::{UVec2, Vec2};
+use bevy_render::camera::Camera;
 use bevy_transform::components::Transform;
 use bevy_utils::HashMap;
-use bevy_window::{Window, WindowId, WindowScaleFactorChanged, Windows};
+use bevy_window::{WindowId, WindowScaleFactorChanged, Windows};
 use std::fmt;
 use taffy::{number::Number, Taffy};
 
 #[derive(Resource)]
 pub struct FlexSurface {
+    /// Maps UI entities to taffy nodes.
     entity_to_taffy: HashMap<Entity, taffy::node::Node>,
-    window_nodes: HashMap<WindowId, taffy::node::Node>,
+    /// Maps UI root entities to taffy root nodes.
+    ///
+    /// The taffy root node is a special node that has only one child: the real root node
+    /// present that is stored in `entity_to_taffy`. This means that two taffy nodes are
+    /// maintained for each bevy_ui root node.
+    root_nodes: HashMap<Entity, taffy::node::Node>,
```

Comment:
Is this a design or limitation already existing or a design choice you have made?

Is there value in doing this special-casing instead of just introducing multiple taffy-trees, one for each new root?